examples/Advanced/Advanced.ino

```diff
@@ -0,0 +1,47 @@
+/*
+    WifiSettings advanced example
+
+    Demonstrates callback functions and custom variables
+    to be saved through WifiSettings.
+
+    Source and further documentation available at
+    https://github.com/Juerd/esp32-WiFiSettings
+*/
+
+#include <SPIFFS.h>
+#include <WiFi.h>
+#include <WiFiSettings.h>
+
+void setup() {
+    Serial.begin(115200);
+    SPIFFS.begin(true);  // Will format on the first run after failing to mount
+
+    // Set custom callback functions
+    WiFiSettings.onSuccess  = []() {
+        Serial.println("Successfully connected to WiFi");
+    };
+    WiFiSettings.onFailure  = []() {
+        Serial.println("Failed to connect to WiFi!");
+    };
+    WiFiSettings.onWaitLoop = []() {
+        Serial.print(".");
+        return 500; // Delay next function call by 1000ms
+    };
+
+    // Callback functions do not have to be lambda's, e.g.
+    // WiFiSettings.onPortalWaitLoop = []() { blink(); };
+
+    // Define custom settings saved by WifiSettings
+    // These will return the default if nothing was set before
+    String host = WiFiSettings.string( "server_host", "default.example.org");
+    // Integers have a min, max, and default
```

Comment:
The min and max are optional (also, you can add minimum and maximum lengths to strings). Maybe an example with both?

Reply:
I'd prefer to leave the examples very simple, otherwise it's a slippery slope to inadvertently documenting all library functions in one sketch.